tests/test_regex.py

```diff
@@ -1,37 +1,267 @@
-import hypothesis_regex
+import hypothesis as h
+import hypothesis.errors as he
+
+from hypothesis_regex import regex, UNICODE_CATEGORIES, UNICODE_DIGIT_CATEGORIES, \
+    UNICODE_SPACE_CATEGORIES, UNICODE_WORD_CATEGORIES, UNICODE_WEIRD_NONWORD_CHARS, \
+    SPACE_CHARS, UNICODE_SPACE_CHARS, HAS_WEIRD_WORD_CHARS
 import pytest
 import re
+import six
 import six.moves
+import sys
+import unicodedata
+
+
+def is_ascii(s):
+    return all(ord(c) < 128 for c in s)
+
+
+def is_digit(s):
+    return all(unicodedata.category(c) in UNICODE_DIGIT_CATEGORIES for c in s)
+
+
+def is_space(s):
+    return all(c in SPACE_CHARS for c in s)
+
+
+def is_unicode_space(s):
+    return all(
+        unicodedata.category(c) in UNICODE_SPACE_CATEGORIES or \
+        c in UNICODE_SPACE_CHARS
+        for c in s
+    )
+
+
+def is_word(s):
+    return all(
+        c == '_' or (
+            (not HAS_WEIRD_WORD_CHARS or c not in UNICODE_WEIRD_NONWORD_CHARS) and
+            unicodedata.category(c) in UNICODE_WORD_CATEGORIES
+        )
+        for c in s
+    )
+
+
+class TestRegexUnicodeMatching:
+    def _test_matching_pattern(self, pattern, isvalidchar, unicode=False):
+        if unicode:
+            r = re.compile(pattern) if six.PY3 else re.compile(pattern, re.UNICODE)
```

Comment:
Although I see no harm in inline conditionals, I will still introduce functions to compile ascii/unicode regexes just to make code more compact.